### PR TITLE
Refactor IntOrString into a new pkg

### PR DESCRIPTION
**Pull request by thockin**
pkg/util/intstr is a cleaner encapsulation for this type and supporting
functions.  No behavioral changes.

This has annoyed me for a long time, and one too many code reviews that touched
it finally put me over the edge.

### CODE REVIEW

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by derekwaynecarr** [comment]
This reads a lot better ;-)

File: pkg/util/intstr/intstr_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (derekwaynecarr) [inline pkg/util/intstr/intstr_test.go]: 2015

File: pkg/util/intstr/intstr.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (derekwaynecarr) [inline pkg/util/intstr/intstr.go]: 2015

**Comment by derekwaynecarr** [comment]
minor nit on copyright, otherwise LGTM

File: pkg/util/intstr/intstr_test.go (hunk 1 above)
Comment (thockin) [inline pkg/util/intstr/intstr_test.go]: The original files were 2014, we're not supposed to update copyrights

File: pkg/util/intstr/intstr.go (hunk 2 above)
Comment (thockin) [inline pkg/util/intstr/intstr.go]: ditto

**Comment by derekwaynecarr** [comment]
@thockin - sorry for the confusion, i thought new files got new header.  needs a rebase.

**Comment by thockin** [comment]
rebased, tests re-running

**Comment by thockin** [comment]
50/50 on shippable

**Comment by derekwaynecarr** [comment]
Ships me has been flaky

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by thockin** [comment]
Was LGTM then rebased, re-adding LGTM

**Comment by thockin** [comment]
Actually I am removing LGTM - this will make cherry-picks hard and I don't think it is worthy of a cherry pick by itself, so it can wait a couple weeks.

**Comment by derekwaynecarr** [comment]
@thockin - feel free to self LGTM tag when / if you want this to go in.

**Comment by thockin** [comment]
new version pushed

**Comment by thockin** [comment]
rebased and merged

**Comment by derekwaynecarr** [comment]
@k8s-bot test this

**Comment by derekwaynecarr** [comment]
Looks like a compile error:
```
[20:12:08][Step 4/4] _output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:36:2: cannot find package "k8s.io/kubernetes/pkg/util/intstr" in any of:
[20:12:08][Step 4/4] 	/usr/src/go/src/k8s.io/kubernetes/pkg/util/intstr (from $GOROOT)
[20:12:08][Step 4/4] 	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/intstr (from $GOPATH)
[20:12:08][Step 4/4] 	/go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/intstr
[20:12:08][Step 4/4] !!! Error in /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:392
[20:12:08][Step 4/4]   'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 1
[20:12:08][Step 4/4] Call stack:
[20:12:08][Step 4/4]   1: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:392 kube::golang::build_binaries_for_platform(...)
[20:12:08][Step 4/4]   2: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:527 kube::golang::build_binaries(...)
[20:12:08][Step 4/4]   3: hack/build-go.sh:26 main(...)
[20:12:08][Step 4/4] Exiting with status 1
[20:12:08][Step 4/4] !!! Error in /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:448
```

**Comment by thockin** [comment]
@k8s-bot test this please

@k8s-bot  unit test this please

**Comment by thockin** [comment]
@ixdy I need an assist.  This is failing because it can't find a package that was added in this PR.

```
13:20:52 +++ [1112 21:20:52] Building go targets for linux/amd64:
13:20:52     cmd/kube-proxy
13:20:52     cmd/kube-apiserver
13:20:52     cmd/kube-controller-manager
13:20:52     cmd/kubelet
13:20:52     cmd/kubemark
13:20:52     cmd/hyperkube
13:20:52     cmd/linkcheck
13:20:52     plugin/cmd/kube-scheduler
13:20:52 _output/dockerized/go/src/k8s.io/kubernetes/pkg/util/util.go:36:2: cannot find package "k8s.io/kubernetes/pkg/util/intstr" in any of:
13:20:52 	/usr/src/go/src/k8s.io/kubernetes/pkg/util/intstr (from $GOROOT)
13:20:52 	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/intstr (from $GOPATH)
13:20:52 	/go/src/k8s.io/kubernetes/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/intstr
13:20:52 !!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:392
13:20:52   'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 1
```

This seems like a failure of test infra - not the PR.  Help?

**Comment by thockin** [comment]
NM, I screwed up.  Always blame the tools first :)  trying again.

**Comment by thockin** [comment]
@k8s-bot unit test this please

**Comment by thockin** [comment]
@k8s-bot unit test this please

**Pull request merged**